src/core/lv2/Lv2Proc.cpp

```diff
@@ -566,21 +600,37 @@ void Lv2Proc::createPort(std::size_t portNum)
 						break;
 					case Lv2Ports::Vis::Enumeration:
 					{
-						auto comboModel = new ComboBoxModel(nullptr, dispName);
-						LilvScalePoints* sps =
-							lilv_port_get_scale_points(m_plugin, lilvPort);
-						LILV_FOREACH(scale_points, i, sps)
+						ComboBoxModel* comboModel = new ComboBoxModel(nullptr, dispName);
+
+						{
+							LilvScalePoints* sps =
+								lilv_port_get_scale_points(m_plugin, lilvPort);
+							// temporary map, since lilv may return scale points in random order
+							std::map<float, const char*> scalePointMap;
+							LILV_FOREACH(scale_points, i, sps)
+							{
+								const LilvScalePoint* sp = lilv_scale_points_get(sps, i);
+								const float f = lilv_node_as_float(lilv_scale_point_get_value(sp));
+								const char* s = lilv_node_as_string(lilv_scale_point_get_label(sp));
+								scalePointMap[f] = s;
+							}
+							for (const auto& [f,s] : scalePointMap)
+							{
+								ctrl->m_scalePointMap.push_back(f);
+								comboModel->addItem(s);
+							}
+							lilv_scale_points_free(sps);
```

Comment:
Manually calling `*_free` functions is a bit of a code smell in C++. This would be better managed by an RAII construct.